code/modules/projectiles/gun.dm

```diff
@@ -636,11 +633,18 @@
 	return launched
 
 /obj/item/weapon/gun/proc/play_fire_sound(var/mob/user, var/obj/item/projectile/P)
-	var/shot_sound = (istype(P) && P.fire_sound)? P.fire_sound : fire_sound

```

Comment:
`var/shot_sound = fire_sound ? fire_sound : ((istype(P) && P.fire_sound)? P.fire_sound : [generic gunshot sound])`
Copypasta bad

Reply:
Welp, it's not copypasta anymore.

Reply:
Actually, on second thought, give me a minute. This line just pisses me off the more I look at it.

Reply:
Use the provided line in place of the highlighted line, and remove the `if(fire_sound) else` block, you just need the one piece of
> if(silenced)
> > [quiet sound]
> else
> > [regular sound]

Reply:
Wait, what? _Why_ would I do that?

Reply:
```if(fire_sound) // If the gun has its own fire-sound, use that.
		if(silenced)
			playsound(user, fire_sound, 10, 1)
		else
			playsound(user, fire_sound, 50, 1)
	else // If it has no special sound, default to the sound that the projectile uses.
		if(P.fire_sound) // Catches possible runtime where fire_sound is null for some reason.
			if(silenced)
				playsound(user, P.fire_sound, 10, 1)
			else
				playsound(user, P.fire_sound, 50, 1)```

The if/else block is the reason this PR exists. If the **gun** has a fire sound, **use that sound**. If that var is null, it will go to the else part, and use the projectile sound.

Reply:
Okay now I get what you're saying after a Discord convo. I got totally lost.